src/Calendar.php

```diff
@@ -444,13 +444,13 @@ private function determineCorrectTopStatusForProjection(): Status
 
     /**
      * This method can determine the top level booking availability from the sub events aka timestamps
-     * - If there are no timestamps this is a periodic or permanent calendar which is always available
+     * - For a periodic or permanent calendar this is always available
      * - If one of the timestamps is available then the top level is available
      * - If all of the timestamps are unavailable the top level is also unavailable
      */
     private function deriveBookingAvailabilityFromSubEvents(): BookingAvailability
     {
-        if (empty($this->timestamps)) {
+        if (!$this->allowsUpdatingBookingAvailability()) {
```

Comment:
I slightly preferred the `if (empty($this->timestamps))` here, because it decouples the write and read side a bit better. The `allowsUpdatingBookingAvailability` is used when updating, and  `deriveBookingAvailabilityFromSubEvents` to determine the current state. It's a problem that we have in other places as well, that we accidentally validate data that is already persisted anyway which can lead to more bugs because the historically persisted data may not always comply to the latest domain rules.

Reply:
Good point! I updated this.